libs/clients/stripe/client.go

```diff
@@ -0,0 +1,144 @@
+package stripe
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/brave-intl/bat-go/libs/clients"
+	appctx "github.com/brave-intl/bat-go/libs/context"
+	"github.com/google/go-querystring/query"
+)
+
+// Client abstracts over the underlying client
+type Client interface {
+	CreateOnrampSession(
+		ctx context.Context,
+		integrationMode string,
+		walletAddress string,
+		sourceCurrency string,
+		sourceExchangeAmount string,
+		destinationNetwork string,
+		destinationCurrency string,
+		supportedDestinationNetworks []string,
+	) (*OnrampSessionResponse, error)
+}
+
+// HTTPClient wraps http.Client for interacting with the Stripe server
+type HTTPClient struct {
+	client *clients.SimpleHTTPClient
+}
+
+// NewWithContext returns a new HTTPClient, retrieving the base URL from the context
+func NewWithContext(ctx context.Context) (Client, error) {
+	// get the server url from context
+	serverURL, err := appctx.GetStringFromContext(ctx, appctx.StripeOnrampServerCTXKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get StripeServer from context: %w", err)
+	}
+
+	// get the server secretKey from context
+	secretKey, err := appctx.GetStringFromContext(ctx, appctx.StripeOnrampSecretKeyCTXKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get StripeSecretKey from context: %w", err)
+	}
+
+	client, err := clients.NewWithHTTPClient(serverURL, secretKey, &http.Client{
+		Timeout: time.Second * 30,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return NewClientWithPrometheus(
+		&HTTPClient{
+			client: client,
+		}, "stripe_onramp_context_client"), nil
+}
+
+// onrampSessionParams for fetching prices
+type onrampSessionParams struct {
+	IntegrationMode              string   `url:"integration_mode"`
+	WalletAddress                string   `url:"-"`
+	SourceCurrency               string   `url:"transaction_details[source_currency],omitempty"`
+	SourceExchangeAmount         string   `url:"transaction_details[source_exchange_amount],omitempty"`
+	DestinationNetwork           string   `url:"transaction_details[destination_network],omitempty"`
+	DestinationCurrency          string   `url:"transaction_details[destination_currency],omitempty"`
+	SupportedDestinationNetworks []string `url:"-"`
+}
+
+// GenerateQueryString - implement the QueryStringBody interface
+func (p *onrampSessionParams) GenerateQueryString() (url.Values, error) {
+	values, err := query.Values(p)
+	if err != nil {
+		return nil, err
+	}
+	if p.WalletAddress != "" {
+		key := fmt.Sprintf("transaction_details[wallet_addresses][%s]", p.DestinationNetwork)
+		values.Add(key, p.WalletAddress)
+	}
+
+	if len(p.SupportedDestinationNetworks) > 0 {
+		for i, network := range p.SupportedDestinationNetworks {
+			key := fmt.Sprintf("transaction_details[supported_destination_networks][%d]", i)
+			values.Add(key, network)
+		}
+	}
+
+	return values, nil
+}
```

Comment:
The Stripe servers expect a POST request with content type "application/x-www-form-urlencoded", not JSON.  This is kind of tricky since it seems Go only really has support for very simple url encoded data structurs, i.e. a map of strings to strings (e.g. param1=a&param2=b...).  Stripe has many nested structures, some of which are lists, so this payload needs to be created manually in this function.